providers/stores/mongo.js

```diff
@@ -126,24 +128,43 @@ class MongoStore {
   async store(definition) {
     const pageSize = 1000
     definition._id = this._getId(definition.coordinates)
-    await this.collection.deleteMany({ '_mongo.partitionKey': definition._id })
     const pages = Math.ceil((get(definition, 'files.length') || 1) / pageSize)
-    const result = await this.collection.insertMany(
+    const slug = uuid()
+    const result = await this.collection.bulkWrite(
       range(pages).map(
         index => {
           if (index === 0) {
             const definitionPage = clone(definition)
             if (definition.files) definitionPage.files = definition.files.slice(0, pageSize)
-            return { ...definitionPage, _mongo: { partitionKey: definition._id, page: 1, totalPages: pages } }
+            return {
+              updateOne: {
+                filter: { _id: definition._id },
```

Comment:
We should update this filter to include

```
'described.tools.length' : { $lt: definition.described.tools.length }
```

This will preserve the order, so we won't write unless the number of tools are increasing.

Reply:
not entirely convinced as the set can be different but the same length. I suppose that would be a lost up date situation but there is no clear winner.

Reply:
good point. this just came up as a mechanism that I wanted to capture while we figure out exactly what to do here